src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Centipede engine interface."""
+
+import glob
+import os
+from pathlib import Path
+import re
+
+from clusterfuzz._internal.bot.fuzzers import dictionary_manager
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.system import new_process
+from clusterfuzz.fuzz import engine
+
+_CLEAN_EXIT_SECS = 10
+_TIMEOUT = 1200
+_SERVER_COUNT = 1
+_RSS_LIMIT = 4096
```

Comment:
Wonder if this should be abstraced somewhere that libFuzzer and centipede can use it (maybe engine_common)

Reply:
Not sure my finding is related: It seems `libFuzzer` reads some of the flags from the [an option file](https://github.com/google/clusterfuzz/blob/388f4cf83daa3aa0695a30de4c0eb446ff828ffd/src/clusterfuzz/_internal/tests/core/bot/tasks/corpus_pruning_task_data/build/test_get_libfuzzer_flags.options#L10).
I reckon this will take a bit of effort, maybe it should be done in a separate PR?